core/blockly.js

```diff
@@ -248,34 +248,35 @@ Blockly.onKeyDown_ = function(e) {
 
 /**
  * Copy a block onto the local clipboard.
- * @param {!Blockly.Block} block Block to be copied.
+ * @param {!Blockly.Block | !Blockly.WorkspaceComment} toCopy Block or Workspace Comment
+ *    to be copied.
  * @private
  */
-Blockly.copy_ = function(block) {
-  var xmlBlock = Blockly.Xml.blockToDom(block);
+Blockly.copy_ = function(toCopy) {
+  var xml = (toCopy.isComment) ? toCopy.toXmlWithXY() : Blockly.Xml.blockToDom(toCopy);
   // Copy only the selected block and internal blocks.
-  Blockly.Xml.deleteNext(xmlBlock);
+  Blockly.Xml.deleteNext(xml);
   // Encode start position in XML.
-  var xy = block.getRelativeToSurfaceXY();
-  xmlBlock.setAttribute('x', block.RTL ? -xy.x : xy.x);
-  xmlBlock.setAttribute('y', xy.y);
-  Blockly.clipboardXml_ = xmlBlock;
-  Blockly.clipboardSource_ = block.workspace;
+  var xy = toCopy.getRelativeToSurfaceXY();
+  xml.setAttribute('x', toCopy.RTL ? -xy.x : xy.x);
```

Comment:
`toXmlWithXY` already does this for a comment.  Let's do

```
if (toCopy.isComment) {
  var xml = toCopy.toXmlWithXy();
} else {
  var xml = Blockly.Xml.blockToDom(toCopy);
  // Copy only the selected block and internal blocks.
  Blockly.Xml.deleteNext(xml);
  // Encode start position in XML.
  var xy = toCopy.getRelativeToSurfaceXY();
  xml.setAttribute('x', toCopy.RTL ? -xy.x : xy.x);
  xml.setAttribute('y', xy.y);
}

Blockly.clipboardXml_ = xml;
Blockly.clipboardSource_ = toCopy.workspace;
```

As a bonus, now we're not calling `deleteNext` on a WS comment, which didn't really make sense anyway.  (But worked because JS).